.circleci/config.yml

```diff
@@ -34,6 +34,9 @@ jobs:
       - deploy:
           name: Deploy
           command: |
+            if [[ -z "$PREVENT_DEPLOY" ]]; then
```

Comment:
Will this allow us to have an empty `PREVENT_DEPLOY` environment variable set in CircleCI or does it have to not exist entirely? The advantage of it existing and empty is that it's clear what needs to be done just by browsing the environment variables.

Reply:
It could also be empty, but circle doesn't allow editing env variables because it masks them, only deleting and creating them.

Reply:
Hmm... okay. My goal is to make it as easy as possible to find the switch. I'd rather not have to read this shell command. One of the wikis might be a good place, though personally I often forget to look there. If I'm looking for a switch, the first place I'm going to look is circleci.com. Any ideas?

Reply:
Great job describing your concern. I actually think that this file is short-enough (~90 lines) that it isn't unreasonable to call it the documentation. I'd change the logic, which I read as, "if `PREVENT_DEPLOY` is blank," to something more like, "if `PREVENT_DEPLOY` is 'yes.'"

Also, note that if you read the Deploying page on the Wiki for production deployments, you'd be directed to this file, which I think is absolutely brilliant.

Reply:
The annoyance here is that when I created the variable and set it to an empty string, it rendered as `xxxx` in the circle UI, although that's really the right thing for it to do... I can't decide if it's clearer to have it absent or set to something blank so that you know it's there. I'm good with requiring it to be set to something to prevent the deploy, although I lean more towards 'true' over 'yes'

Reply:
Just throwing my 2 cents in with a thought... I am not sure what the motivation is behind this, but I am guessing it is a last ditch effort to stop a deploy in case a bug was noticed very close to the deploy time where rolling back the offending change would take too long for various reasons.

What about adding a file flag of some sort - something like a `STOP_DEPLOY`.json file? The contents of the file could be: `{stopDeploy: false}`. The deploy step could run a check against this. If a developer needed to stop a deploy, change the value to true and merge the PR. For this special action, no approval would be necessary. Although this would add a bit of extra latency to the circle process.

This may be more straightforward for anyone to stop the deploy and could prevent a situation where the flag is accidentally left on - or at least we would know how long it was left on for. It would also allow a certain degree of auditing - you could go through PR's and see how many times a stop deploy took place, when and figure out why, even months later.

Reply:
@petesantos This is more concerned with preventing deploys when there's a request to do so for some reason, like a workshop, so that we don't have to avoid merging PR's. The recent incident was when there was a problem with the monolith prod deploy, and we were asked to suspend our deploy just to keep in sync.